tests/test_credentials.py

```diff
@@ -3,15 +3,20 @@
 
 
 @pytest.mark.parametrize(
-    "resource_type_method,client_type",
+    "resource_type,client_type",
     [
-        ("get_app_client", AppsV1Api),
-        ("get_batch_client", BatchV1Api),
-        ("get_core_client", CoreV1Api),
+        ("apps", AppsV1Api),
+        ("batch", BatchV1Api),
+        ("core", CoreV1Api),
     ],
 )
-def test_client_return_type(kubernetes_credentials, resource_type_method, client_type):
-    resource_specific_client = getattr(kubernetes_credentials, resource_type_method)
+def test_client_return_type(kubernetes_credentials, resource_type, client_type):
 
-    with resource_specific_client() as client:
+    with kubernetes_credentials.get_client(resource_type) as client:
         assert isinstance(client, client_type)
+
+
+def test_client_bad_resource_type(kubernetes_credentials):
+    with pytest.raises(ValueError):
```

Comment:
Can you pass an argument for `match` to check that the message for the raised exception is matches expectations?
```
    with pytest.raises(ValueError, match="..."):
```